### PR TITLE
[testing] add unit test cases for questions_bloc

### PR DESCRIPTION
**Pull request by felangel**
https://github.com/coronavirus-diary/coronavirus-diary/pull/52#discussion_r396592204
- Add `mockito` and `bloc_test` as `dev_dependencies`
- Add unit tests for `questions_bloc`

Feel free to close this if you prefer not to add the 2 dev_dependencies.

### CODE REVIEW

File: test/blocs/questions_bloc_test.dart [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import 'package:bloc_test/bloc_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+import 'package:covidnearme/src/blocs/questions/questions.dart';
+import 'package:covidnearme/src/data/repositories/questions.dart';
+
+class MockQuestionsRepository extends Mock implements QuestionsRepository {}
```
Comment (johnsonmh) [inline test/blocs/questions_bloc_test.dart:8]: Would it be crazy to use the real `QuestionsRepository`? Or will that at some point be retrieving data over the wire or something?

**Review by johnsonmh — COMMENTED** [review]
LGTM with one question

File: test/blocs/questions_bloc_test.dart (hunk 1 above)
Comment (felangel) [reply test/blocs/questions_bloc_test.dart:8]: Great question! If we used the real `QuestionsRepository` we'd be making real requests and would not be able to easily test the success and failure scenario. We could definitely add integration tests to test the whole flow end to end 👍

**Review by jonahwilliams — COMMENTED** [review]
This PR adds a lot of fixtures to the tests. This might simplify some of the setup initially, but it has the disadvantage of increasing the amount of code that needs to be both read to understand and run during the test. Since tests themselves aren't tested, less code is usually better - and simpler code, even code that isn't that DRY, is better than what would normally qualify as clean code.

**Comment by felangel** [comment]
> This PR adds a lot of fixtures to the tests. This might simplify some of the setup initially, but it has the disadvantage of increasing the amount of code that needs to be both read to understand and run during the test. Since tests themselves aren't tested, less code is usually better - and simpler code, even code that isn't that DRY, is better than what would normally qualify as clean code.

That's fair, I was just putting this out there because in my experience it makes writing unit tests easier and handles some false-positives which are possible with `emitsInOrder`. For what it's worth `bloc_test` itself is tested but if you prefer to omit it that's totally fine by me. Feel free to close this PR if you don't feel it's beneficial.

File: test/blocs/questions_bloc_test.dart (hunk 1 above)
Comment (ferndot) [reply test/blocs/questions_bloc_test.dart:8]: The real questions repository will be pulling from the API in a future version.

**Pull request closed without merging**